src/cbl-mariner/2.0/crossdeps-builder/Dockerfile

```diff
@@ -0,0 +1,64 @@
+FROM mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0-crossdeps-local
+
+# TODO: remove this once debootstrap is available in the base repos.
+COPY mariner-extended.repo /etc/yum.repos.d/
+
+RUN tdnf install -y \
+    # Rootfs build dependencies
+    debootstrap \
+    # LLVM build dependencies
+    texinfo \
+    diffutils \
+    binutils
+
+# Obtain ubuntu package signing key (for use by debootstrap)
+# 1. Add public key used to sign the ubuntu keyring
+COPY dimitri_john_ledkov.asc .
+RUN gpg --output dimitri_john_ledkov.gpg --dearmor dimitri_john_ledkov.asc && \
+    rm dimitri_john_ledkov.asc && \
+# 2. Download the ubuntu keyrings
+    wget https://mirrors.edge.kernel.org/ubuntu/pool/main/u/ubuntu-keyring/ubuntu-keyring_2021.03.26.tar.gz && \
+    echo "492eed5c06408c6f632577adb0796130af5d6542013ef418f47187a209e49bb1 ubuntu-keyring_2021.03.26.tar.gz" | sha256sum -c && \
+    tar xf ubuntu-keyring_2021.03.26.tar.gz && \
+    rm ubuntu-keyring_2021.03.26.tar.gz && \
+# 3. Verify keyrings
+    pushd ubuntu-keyring-2021.03.26 && \
+    gpg --keyring /dimitri_john_ledkov.gpg --output SHA512SUMS.txt --decrypt SHA512SUMS.txt.asc && \
+    rm /dimitri_john_ledkov.gpg && \
+    sha512sum -c SHA512SUMS.txt && \
+# 4. Install the needed keyring and delete the rest
+    mkdir -p /usr/share/keyrings && \
+    mv keyrings/ubuntu-archive-keyring.gpg /usr/share/keyrings && \
+    popd && \
+    rm -r ubuntu-keyring-2021.03.26
+
+# 1. Obtain signing keys used to sign llvm sources
+RUN wget https://releases.llvm.org/release-keys.asc && \
+    echo "f181a90697e3ea4b7782f1ee48314a570aef058505b4f3a0ab0611094ec13241 release-keys.asc" | sha256sum -c && \
+    gpg --output release-keys.gpg --dearmor release-keys.asc && \
+    rm release-keys.asc && \
+# 2. Download llvm sources and signature, and verify signature
+    LLVM_VERSION=12.0.1 && \
+    wget -O llvm-project.src.tar.xz.sig https://github.com/llvm/llvm-project/releases/download/llvmorg-${LLVM_VERSION}/llvm-project-${LLVM_VERSION}.src.tar.xz.sig && \
+    echo "6cc956d622a7d3d746de0d71d8ca616a6c291e2c561703ac7a9535f38b999955 llvm-project.src.tar.xz.sig" | sha256sum -c && \
+    wget -O llvm-project.src.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-${LLVM_VERSION}/llvm-project-${LLVM_VERSION}.src.tar.xz && \
+    echo "129cb25cd13677aad951ce5c2deb0fe4afc1e9d98950f53b51bdcfb5a73afa0e llvm-project.src.tar.xz" | sha256sum -c && \
+    gpg --keyring /release-keys.gpg --verify llvm-project.src.tar.xz.sig && \
+    rm llvm-project.src.tar.xz.sig
+
+# Build LLVM cross-toolchain (with support for targeting arm architectures)
+RUN mkdir llvm-project.src && \
```

Comment:
nit: I think if we `make` everything under `/tmp` then the intermediate artifacts will be cleaned up (and we will only have `make install`'d binaries/files under `/usr` in the final image; which is what we need).

Reply:
Cool. If this layer is always going to be consumed in multistage, then no need to switch to /tmp. BTW, all builttools images have empty /tmp, so anything we are putting there won't be persisted.

Reply:
There's nothing special about the /tmp directory. Anything placed there will be committed to the image layer after executing a `RUN` instruction.

Reply:
You are right, we do clear it manually. e.g. https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/e1bc4b7f52f903020b147ee26de6e3e918f67915/src/ubuntu/22.04/cross/x86-alpine/Dockerfile#L3-L20

I was reading a post in docker repo about `/var/tmp` which wasn't get cleaned up as it was supposed to (that bug was fixed in some version of docker).